mewarpx/mewarpx/setups_store/diode_setup.py

```diff
@@ -167,24 +167,27 @@ class DiodeRun_V1(object):
     # steps between doing load-balancing. If <1 load balancing will not be done
     LOAD_BALANCE_INTERVALS = 0
 
-    def __init__(self, dim=1, rz=False, **kwargs):
+    def __init__(self, **kwargs):
         for kw in list(kwargs.keys()):
             setattr(self, kw, kwargs[kw])
 
-        self.dim = dim
+        self.dim = mwxrun.dim
         if self.dim not in [1, 2, 3]:
             raise ValueError(f"Unavailable dimension {self.dim}")
-        self.rz = rz
+        self.rz = mwxrun.geom_str == 'RZ'
         if self.dim != 2 and self.rz:
             raise ValueError("dim=2 required for RZ")
```

Comment:
Based on the direction David went with the `libwarpx` refactor I don't see us having a way to avoid calling `mwxutil.init_libwarpx` anytime soon. I like the changes here in that it unifies specifying the simulation dimension to a single place and therefore avoids the possibility to have different parts of the code think different dimensions are used.